### PR TITLE
Added Fighters tech tree, updated AI

### PR DESCRIPTION
**Pull request by MatGB**
~First pass at balanced fighter tech tree
~Provisional changes to AI dependencies and techlists

### CODE REVIEW

**Comment by MatGB** [comment]
@freeorion/ai-team I think I've understood @Morlic-fo 's changes to dependencies and I *hope* I've made the required adjustments correctly, it plays fine and once I'd figured out my mistakes I didn't get any errors in the logs or crashes as a result.

Can you give them a quick look over to confirm it's all OK? I need to log off, if it's fine I'll get the stringtables done tonight so it can merge in and we can then merge the branch into master.

**Comment by Morlic-fo** [comment]
Dependencies changes look right, good job.

For techlists, it is enough for the techs to be researched. It changes the tech order somewhat (the last weapon upgrade will be delayed until the entire tech group is finished), but that is fine for now. That can be adjusted along with the position of the fighter techs once it is clear when it is a good idea to research it...

File: default/scripting/starting_unlocks/items.inf [hunk 1]
```diff
@@ -6,11 +6,6 @@ Item type = ShipPart name = "FU_BASIC_TANK"
 Item type = ShipPart name = "CO_OUTPOST_POD"
 Item type = ShipPart name = "CO_COLONY_POD"
 Item type = ShipPart name = "DT_DETECTOR_1"
-Item type = ShipPart name = "FT_BAY_1"
-Item type = ShipPart name = "FT_HANGAR_1"
-Item type = ShipPart name = "FT_HANGAR_2"
-Item type = ShipPart name = "FT_HANGAR_3"
-Item type = ShipPart name = "SR_WEAPON_0_1"
```
Comment (Morlic-fo) [inline default/scripting/starting_unlocks/items.inf:13]: Which tech does unlock the Flak Cannon now?

Comment (MatGB) [reply default/scripting/starting_unlocks/items.inf:13]: Self defence, same as Mass Driver, not sure it'll stay there but think it likely and I want to redo the premades so there are some early game Destroyer class ships

Comment (MatGB) [reply default/scripting/starting_unlocks/items.inf:13]: Gah, good spot, it's in a different commit, I'll fix that later.

**Comment by MatGB** [comment]
K, all merged in with errors corrected, VERY basic stringtable content, we'll need to expand on this with a proper Pedia article and similar, but some of that can wait until testing and further balancing is done: the numbers play OK at the moment but we'll probably want to tweak a fair few things as we go.

**Pull request merged**